### PR TITLE
Fixed #28073 -- RemoveField.state_forwards() gives AttributeError.

### PR DESCRIPTION
**Pull request by pallav10**
Added a check if old_field exist, otherwise delay is set to False

### CODE REVIEW

File: django/db/migrations/operations/fields.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def state_forwards(self, app_label, state):
                 old_field = instance
         state.models[app_label, self.model_name_lower].fields = new_fields
         # Delay rendering of relationships if it's not a relational field
-        delay = not old_field.is_relation
+        delay = not old_field.is_relation if old_field else False
```
Comment (charettes) [inline django/db/migrations/operations/fields.py]: Under which circumstances would `old_field` be `None` here? Could you write a test for that?

Comment (pallav10) [reply django/db/migrations/operations/fields.py]: I could find only case if it tries to remove an old_field which does not exist in the model anymore. 
Test added.

**Comment by charettes** [comment]
@pallav10 unless you can get the auto-detector to generate such a series of operations I'm afraid this is not fixing the right thing. Adding test for such a scenario would be testing that the operation behaves properly when the auto-detector doesn't.

I'm afraid the patch proposed here will just hide another issue; you shouldn't be in a state where you're trying to remove a field that is missing.

From a quick look on the ticket it looks like this is only encountered when users manually edit migrations. If they want to run a `ALTER TABLE DROP COLUMN` without altering the existing state they should likely use `SeparateDatabaseAndState([RemoveField], [])` instead.

**Comment by pallav10** [comment]
> @pallav10 unless you can get the auto-detector to generate such a series of operations I'm afraid this is not fixing the right thing. Adding test for such a scenario would be testing that the operation behaves properly when the auto-detector doesn't.
> 
> I'm afraid the patch proposed here will just hide another issue; you shouldn't be in a state where you're trying to remove a field that is missing.
> 
> From a quick look on the ticket, it looks like this is only encountered when users manually edit migrations. If they want to run a `ALTER TABLE DROP COLUMN` without altering the existing state they should likely use `SeparateDatabaseAndState([RemoveField], [])` instead.

In Newer version it seems to be not reproducible, I agree as unless there's a case where the user manually edits the file for an already removed field in DB, this error won't occur. 
Even though from the description in the ticket, people migrating from 1.10 to 1.11 or 2.X faces an issue if one of the migration contains `RemoveField` . Regardless, this patch suggested is to simply remove the incautious even in the case of manual override to migration. 
I will surely try to get the auto-detector to generate such an operation.

Thanks. :)

**Comment by pallav10** [comment]
> `I'm afraid the patch proposed here will just hide another issue; you shouldn't be in a state where you're trying to remove a field that is missing.`

Yes, I agree. Although in such case it should raise the exception `django.core.exceptions.FieldDoesNotExist`

I was able to reproduce this by sample project with the following steps:

1. remove field from model, which creates a migration file on makemigration containing:
`# Generated by Django 2.2.5 on 2020-01-16 11:05

from django.db import migrations


class Migration(migrations.Migration):

    dependencies = [
        ('shops', '0005_auto_20200113_0650'),
    ]

    operations = [
        migrations.RemoveField(
            model_name='customer',
            name='first_name',
        ),
    ]`

2. Modify the migration file manually, change name parameter:
` migrations.RemoveField(
            model_name='customer',
            name='foo',
        ),`

3. Apply the migration, Produces incorrect error:
`File "/home/pallav/.virtualenvs/sample/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 172, in state_forwards
    delay = not old_field.is_relation
AttributeError: 'NoneType' object has no attribute 'is_relation'
`

4. On applying the patch, Corrects the error as:
` File "/home/pallav/.virtualenvs/sample/lib/python3.7/site-packages/django/db/models/options.py", line 567, in get_field
    raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: Customer has no field named 'foo'
`

**Pull request closed without merging**